cpp/src/Slice/Grammar.h

```diff
@@ -43,6 +43,26 @@
 #if YYDEBUG
 extern int slice_debug;
 #endif
+/* "%code requires" blocks.  */
+#line 12 "src/Slice/Grammar.y"
+
+
+// Define a custom location type for storing the location (and filename) of matched tokens.
+#define YYLTYPE Slice::TokenContext
+
+// I must set the initial stack depth to the maximum stack depth to
+// disable bison stack resizing. The bison stack resizing routines use
+// simple malloc/alloc/memcpy calls, which do not work for the
+// YYSTYPE, since YYSTYPE is a C++ type, with constructor, destructor,
+// assignment operator, etc.
+#define YYMAXDEPTH  10000
+#define YYINITDEPTH YYMAXDEPTH
+
+// Newer bison versions allow to disable stack resizing by defining yyoverflow.
+#define yyoverflow(a, b, c, d, e, f, g, h) yyerror(a)
```

Comment:
> // I must set the initial stack depth to the maximum stack depth to
>  // disable bison stack resizing.
> ...
> // Newer bison versions allow to disable stack resizing by defining yyoverflow.

Are these comments conflicting? I assume we're using the latest version. Does that mean we could only define `yyoverlow`, or do we need to also define `YYMAXDEPTH` and `YYINITDEPTH`.

Reply:
This is old code that I kept just to be on the safe side. It seems like the comments are conflicting, but the documentation was pretty vague and lacking, so I couldn't be entirely certain.
Either way, I'm planning on switching Bison and Flex to use a C++ mapping, at which point all of this will be removed anyways. So I left it for the time at hand.